scripts/linters/js_ts_linter.py

```diff
@@ -429,6 +436,82 @@ def _check_angular_services_index(self) -> concurrent_task_utils.TaskResult:
         return concurrent_task_utils.TaskResult(
             name, failed, error_messages, error_messages)
 
+    def _check_unknown_type(self) -> concurrent_task_utils.TaskResult:
+        """Prints a list of lint errors if an unknown type is used. Add proper
+        comment before the line with the unknown type if the type is needed.
+
+        Returns:
+            TaskResult. A TaskResult object representing the result of the
+            lint check.
+        """
+        name = 'Unknown type'
+        error_messages: List[str] = []
+        failed = False
+        ts_files_to_check = self.ts_filepaths
+        for file_path in ts_files_to_check:
+            comment_before_unknown_type = False
+            multiple_line_comment = False
+            # Not showing lint errors for files present in
+            # FILES_CONTAINING_UNKNOWN_TYPE.
+            if file_path in FILES_CONTAINING_UNKNOWN_TYPE:
+                continue
+
+            file_content = self.file_cache.read(file_path)
+            for line_num, line in enumerate(file_content.split('\n')):
+                unknown_type_indexes: List[int] = []
+                # Check if the line has a comment. Skip the line if it has
+                # a comment.
+                if not (line.strip().startswith('//') or multiple_line_comment):
+                    # Allow the use of 'as unknown' type conversion in spec
+                    # files.
+                    if 'spec' in file_path:
```

Comment:
This is too broad of a condition. For example, what about `spectacular.ts` contains `spec` but is not a spec file